firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexBackfiller.java

```diff
@@ -201,6 +235,40 @@ private IndexOffset getNewOffset(List<Document> documents, IndexOffset currentOf
     return oldestDocuments.subList(0, Math.min(count, oldestDocuments.size()));
   }
 
+  /**
+   * Returns up to {@code count} documents sorted by batch id ascending, except in cases where the
+   * mutation batch id contains more documents. This method will always return all documents in a
```

Comment:
It sounds like all bets are off when the batch ID contains more documents - maybe rewrite the intro to make it more clear that this only applies to the count.